src/meta/src/hummock/model/key_range.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp;
+
+use risingwave_hummock_sdk::VersionedComparator;
+use risingwave_pb::hummock::KeyRange;
+
+/// Implements a subset methods of `hummock_sdk::KeyRange` for prost `KeyRange`
+pub trait KeyRangeExt {
```

Comment:
Could we also implement `KeyRangeExt` for `hummock_std::KeyRange` ? 
I means that we can make a macro for both `hummock_std::KeyRange` and `meta::KeyRange` so that we can keep code always same

Reply:
> Could we also implement `KeyRangeExt` for `hummock_std::KeyRange` ? I means that we can make a macro for both `hummock_std::KeyRange` and `meta::KeyRange` so that we can keep code always same

Yes macro. Let me fix it.

Reply:
Added 2 macro:

- `impl_key_range_common` to implement `full_key_overlap` and `full_key_extend`.
- `key_range_cmp` to perform comparison.

PTAL